docs/content/dagster-plus/deployment/serverless.mdx

```diff
@@ -306,11 +306,20 @@ After changing the deployment type, you will need to update your code locations'
 
 Unlike Hybrid, Serverless Deployments on Dagster+ require direct access to your data, secrets and source code.
 
-- Dagster+ Serverless does not provide persistent storage. Ephemeral storage is deleted when a run concludes.
 - Secrets and source code are built into the image directly. Images are stored in a per-customer container registry with restricted access.
 - User code is securely sandboxed using modern container sandboxing techniques.
 - All production access is governed by industry-standard best practices which are regularly audited.
 
+### IO Management on Serverless
+
+<Warning>
+If you are a Serverless customer who is working on personally identifiable information (PII), private health information (PHI), has signed a business association agreement (BAA), or are otherwise working with data subject to GDPR or other such regulations, you must not use the default IO manager.
+</Warning> 
```

Comment:
```suggestion
<Note>
The default I/O manager cannot be used if you are a Serverless user who:
<ul>
<li>Works with personally identifiable information (PII)</li>
<li>Works with private health information (PHI)</li>
<li>Has signed a business association agreement (BAA), or</li>
<li>Are otherwise working with data subject to GDPR or other such regulations</li>
</ul>
</Note> 
```

A few suggestions here:

I know `Warning` is techincally the correct admonition here, but our current styling has the callout blend into the page:

![Screenshot 2024-06-14 at 11 57 50 AM](https://github.com/dagster-io/dagster/assets/16615212/af7ea6d8-59e5-4881-b26f-2c3321f17fe9)

Given how important this is, I think using `Note` to make this stand out is a good idea.

I also suggest we lead with "You can't use the default I/O manager" so the thing we're warning people about comes first. I also put the criteria into a list to make it skimmable. (I know the HTML is ugly, but soon we won't have to do this in callouts!)

Thoughts?